torch/testing/_internal/common_methods_invocations.py

```diff
@@ -44,12 +44,14 @@ def __init__(self, cls_name=None, test_name=None, *,
 class SampleInput(object):
     """Represents sample inputs to a function."""
 
-    __slots__ = ['input', 'args', 'kwargs']
+    # output_process_fn_grad is a function that modifies the output of op compatible with input
+    __slots__ = ['input', 'args', 'kwargs', 'output_process_fn_grad']
 
-    def __init__(self, input, *, args=tuple(), kwargs=None):
+    def __init__(self, input, *, args=tuple(), kwargs=None, output_process_fn_grad=None):
```

Comment:
https://github.com/pytorch/pytorch/pull/48627 implemented similar functionality that I think this will be able to reuse. It should land soon.

Reply:
I don't see how to reuse it. I need functionality to be able for the specific inputs define specific function to test by gradcheck. So that not `torch.svd(input)` is tested but `output_process_fn_grad(torch.svd(input))`.

Reply:
Okay, I see `op.output_func` that does the same, but to use it here I'd need separate tests in `op_db` list. I think it's not possible to define different OpInfos for the same function because everything is tied to the `op.name` that has to be unique now.